templates/statefulset.yaml

```diff
@@ -29,11 +32,72 @@ spec:
 {{- if .Values.nodeSelector }}
       nodeSelector:
 {{ toYaml .Values.nodeSelector | indent 8 }}
+{{- end }}
+      securityContext:
+        runAsUser: 1000
+        fsGroup: 1000
+      initContainers:
+{{- if .Values.ca.enabled }}
+      - name: {{ template "apache-nifi.name" . }}-cert
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        command:
+        - bash
+        - -c
+        - |
+          CA_ADDRESS="{{ template "apache-nifi.fullname" . }}-ca:{{ .Values.ca.port }}"
+          until echo "" | openssl s_client -connect "${CA_ADDRESS}"; do
+            echo "Waiting for CA to be avaiable at ${CA_ADDRESS}"
+            sleep 2
+          done;
+
+          mkdir -p ${NIFI_BASE_DIR}/data/cert
+          cd ${NIFI_BASE_DIR}/data/cert
+
+          rm -f ${NIFI_BASE_DIR}/data/cert/*
+
+          ${NIFI_TOOLKIT_HOME}/bin/tls-toolkit.sh client\
+            -c "{{ template "apache-nifi.fullname" . }}-ca"\
+            -t "{{ .Values.ca.token }}"\
+            -p {{ .Values.ca.port }}
+        volumeMounts:
+          - name: "data"
+            mountPath: /opt/nifi/data
 {{- end }}
       containers:
       - name: {{ template "apache-nifi.name" . }}-server
         imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        command:
+        - sh
+        - -ce
+        - |
+          prop_replace () {
+            target_file=${NIFI_HOME}/conf/nifi.properties
+            echo 'replacing target file ' ${target_file}
+            sed -i -e "s|^$1=.*$|$1=$2|"  ${target_file}
+          }
+
+          FQDN=$(hostname -f)
+
+          cat "${NIFI_HOME}/conf/nifi.temp" > "${NIFI_HOME}/conf/nifi.properties"
+
+          prop_replace nifi.web.http.host ${FQDN}
+          prop_replace nifi.web.https.host ${FQDN}
```

Comment:
We cannot set both nifi.web.http.host and nifi.web.https.host - we must set only https.host if we expect user to login via https. Otherwise, we must set only http.host. If we set both, nifi redirects to the login page on http, which causes and error. In the next version of NiFi, the server won't even start if both of these properties are configured. So we can probably just set nifi.web.https.host to ${FQDN} below, in the "if .Values.ca.enabled" section and also there set nifi.web.http.host to empty string.